File: Backend/src/main/java/com/soen390/flightcrew/model/Building.java
package com.soen390.flightcrew.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class Building {

    @JsonProperty("Campus")
    private String campus;

    @JsonProperty("Building")
    private String buildingCode;

    @JsonProperty("Building_Name")
    private String buildingName;

    @JsonProperty("Building_Long_Name")
    private String buildingLongName;

    @JsonProperty("Address")
    private String address;

    @JsonProperty("Latitude")
    private Double latitude;

    @JsonProperty("Longitude")
    private Double longitude;
}

File: mobile/src/types/Building.ts
export interface Building {
  campus: string;
  buildingCode: string;
  buildingName: string;
  buildingLongName: string;
  address: string;
  latitude: number;
  longitude: number;
  structureType: StructureType;
  description?: string;
  polygons?: { latitude: number; longitude: number }[][];
  Google_Place_Info?: {
    displayName: { text: string };
  };
}

export enum StructureType {
  Building = "BUILDING",
  Grounds = "GROUNDS",
  Point = "POINT",
}

export type Campus = "SGW" | "LOY";
